java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -187,23 +188,22 @@
  */
 JNIEXPORT jobject JNICALL Java_ai_onnxruntime_OnnxTensor_getBuffer
         (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+    (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
     const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtTensorTypeAndShapeInfo* info;
-    checkOrtStatus(jniEnv,api,api->GetTensorTypeAndShape((OrtValue*) handle, &info));
-    size_t arrSize;
-    checkOrtStatus(jniEnv,api,api->GetTensorShapeElementCount(info,&arrSize));
-    ONNXTensorElementDataType onnxTypeEnum;
-    checkOrtStatus(jniEnv,api,api->GetTensorElementType(info,&onnxTypeEnum));
-    api->ReleaseTensorTypeAndShapeInfo(info);
+    OrtValue* ortValue = (OrtValue *) handle;
+    JavaTensorTypeShape typeShape = getTensorTypeShape(jniEnv, api, ortValue);
 
-    size_t typeSize = onnxTypeSize(onnxTypeEnum);
-    size_t sizeBytes = arrSize*typeSize;
+    if (typeShape.valid) {
+      size_t typeSize = onnxTypeSize(typeShape.onnxTypeEnum);
+      size_t sizeBytes = typeShape.arrSize*typeSize;
 
-    uint8_t* arr;
-    checkOrtStatus(jniEnv,api,api->GetTensorMutableData((OrtValue*)handle,(void**)&arr));
+      uint8_t* arr = NULL;
+      checkOrtStatus(jniEnv, api, api->GetTensorMutableData((OrtValue*)handle, (void**)&arr));
```

Comment:
>checkOrtStatus [](http://example.com/codeflow?start=6&length=14)

The return code is not checked. Is it legal to call NewDirectByteBuffer() after an error?

Reply:
The capacity part of it is the issue, so I'll fix this.